File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserAnimalCommentRes.java
package com.team05.petmeeting.domain.user.dto.profile;


import com.team05.petmeeting.domain.comment.entity.AnimalComment;

import java.time.LocalDateTime;
import java.util.List;

public record UserAnimalCommentRes(
        long totalCommentCount,
        List<AnimalCommentItem> comments
) {
    public static UserAnimalCommentRes of(long totalCommentCount, List<AnimalComment> commentList) {
        List<AnimalCommentItem> items = commentList.stream()
                .map(AnimalCommentItem::from)
                .toList();

        return new UserAnimalCommentRes(totalCommentCount, items);
    }

    public record AnimalCommentItem(
            long feedId,
            String desertionNo,
            String content,
            LocalDateTime createdAt,
            LocalDateTime updatedAt
    ) {

        public static AnimalCommentItem from(AnimalComment comment) {
            return new AnimalCommentItem(
                    comment.getAnimal().getId(),
                    comment.getAnimal().getDesertionNo(),
                    comment.getContent(),
                    comment.getCreatedAt(),
                    comment.getUpdatedAt()
            );
        }
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserAnimalCommentRes.kt
package com.team05.petmeeting.domain.user.dto.profile

import com.team05.petmeeting.domain.comment.entity.AnimalComment
import java.time.LocalDateTime


data class UserAnimalCommentRes(
    val totalCommentCount: Long,
    val comments: MutableList<AnimalCommentItem?>
) {
    data class AnimalCommentItem(
        val feedId: Long,
        val desertionNo: String,
        val content: String,
        val createdAt: LocalDateTime,
        val updatedAt: LocalDateTime
    ) {
        companion object {
            @JvmStatic
            fun from(comment: AnimalComment): AnimalCommentItem {
                return AnimalCommentItem(
                    comment.animal.id,
                    comment.animal.desertionNo,
                    comment.content,
                    comment.createdAt,
                    comment.updatedAt
                )
            }
        }
    }

    companion object {
        @JvmStatic
        fun of(totalCommentCount: Long, commentList: MutableList<AnimalComment>): UserAnimalCommentRes {
            val items = commentList.stream()
                .map { comment: AnimalComment -> AnimalCommentItem.from(comment) }
                .toList()

            return UserAnimalCommentRes(totalCommentCount, items)
        }
    }
}